service/history/api/startworkflow/api.go

```diff
@@ -398,11 +399,12 @@ func (s *Starter) resolveDuplicateWorkflowID(
 		currentWorkflowConditionFailed.RunID,
 	)
 
+	newRunID := uuid.NewString()
```

Comment:
This is a notable change. I made it because in my next PR, I'll make the creation of workflow lease configurable. Then MultiOps can create a new lease with a workflow context via the `workflowConsistencyChecker`.